superset/assets/javascripts/SqlLab/components/SqlEditorLeftBar.jsx

```diff
@@ -144,15 +169,25 @@ class SqlEditorLeftBar extends React.PureComponent {
             />
           </div>
           <div className="m-t-5">
-            <Select
+            {this.props.queryEditor.schema && <Select
               name="select-table"
               ref="selectTable"
               isLoading={this.state.tableLoading}
+              value={this.state.tableName}
               placeholder={`Add a table (${this.state.tableOptions.length})`}
               autosize={false}
               onChange={this.changeTable.bind(this)}
               options={this.state.tableOptions}
-            />
+            />}
+            {!this.props.queryEditor.schema && <Select.Async
```

Comment:
not sure if the linter will catch this, but i think it's cleaner to format blocks like this:

```jsx
{!this.props.queryEditor.schema && 
  <Select.Async
    ...
  />
}
```